public/app/plugins/datasource/opentsdb/datasource.js

```diff
@@ -208,6 +208,13 @@ function (angular, _, dateMath) {
         dps.push([v, k * 1000]);
       });
 
+      target.tagKeys = Object.keys(md.tags);
```

Comment:
storing query results (tags in this case) in the query object is not so good. When new tags appear this will change the persisted query model (and cause unsaved changes warning)